### PR TITLE
Document: Added documentation for user public key update methods.

### PR DESCRIPTION
**Pull request by sreepuramsudheer**
Signed-off-by: sreepuramsudheer <ssudheer@progress.com>

### Description
This PR is to document different methods to update user public key in chef server.
[Please describe what this change achieves]

### Issues Resolved
https://chefio.atlassian.net/browse/INFS-137
[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussions that are relevant]

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

File: docs-chef-io/content/server/auth.md [hunk 1]
```diff
@@ -386,6 +386,26 @@ common causes:
...
+Chef infra server user can update public_key using below methods.
+-   Using reregister method
+    Use the reregister argument in knife command to regenerate an RSA key pair for a user. The public key will be stored on the Chef Infra Server and the private key will be displayed on STDOUT or written to a named file.
+    
+    Syntax:- knife user reregister USER_NAME (options)
+    Ex:- knife user reregister "Arjun Koch" 
+
+-  Using knife user key command
+    User can have multiple public keys associated with his account in chef server, each public key have a name and expiration date associated with it.
+    Name is an ascii string used to identify public key, expiration date for the public key is specified as an ISO 8601 formatted string: YYYY-MM-DDTHH:MM:SSZ.
```
Comment (vinay-satish) [inline docs-chef-io/content/server/auth.md:400]: *ASCII

File: docs-chef-io/content/server/auth.md [hunk 2]
```diff
@@ -386,6 +386,61 @@ common causes:
     minutes. This can be fixed by syncing the clock with an Network Time
     Protocol (NTP) server.
 
+## User public key update
+
+Chef infra server user can update public_key using below methods.
```
Comment (lbakerchef) [inline docs-chef-io/content/server/auth.md:391]:
```suggestion
Chef infra server user can update public_key using the following methods.
```

File: docs-chef-io/content/server/auth.md [hunk 3]
```diff
@@ -386,6 +386,61 @@ common causes:
...
+-   Using reregister method
+    Use the reregister argument in knife command to regenerate an RSA key pair for a user. The public key will be stored on the Chef Infra Server and the private key will be displayed on STDOUT or written to a named file.
```
Comment (lbakerchef) [inline docs-chef-io/content/server/auth.md:393]:
```suggestion
    Use the reregister argument in the knife command to regenerate an RSA key pair for a user. The public key will be stored on the Chef Infra Server and the private key will be displayed on STDOUT or written to a named file.
```

File: docs-chef-io/content/server/auth.md (hunk 2 above)
Comment (lbakerchef) [inline docs-chef-io/content/server/auth.md:391]:
```suggestion
Chef infra server user can update public_key using below methods.
```
Should it be public_key or public key?  I don't know the answer, I just noticed that elsewhere here you use public key, not public key.

File: docs-chef-io/content/server/auth.md [hunk 4]
```diff
@@ -386,6 +386,61 @@ common causes:
...
+    
+    Syntax:- knife user reregister USER_NAME (options)
+    Ex:- knife user reregister "Arjun Koch" 
+
+-  Using knife user key command
+    User can have multiple public keys associated with his account in chef server, each public key have a name and expiration date associated with it.
```
Comment (lbakerchef) [inline docs-chef-io/content/server/auth.md:399]:
```suggestion
    The user can have multiple public keys associated with his chef server account.  Each public key has a name and expiration date associated with it.
```

File: docs-chef-io/content/server/auth.md [hunk 5]
```diff
@@ -386,6 +386,61 @@ common causes:
...
+    Name is an ascii string used to identify public key, expiration date for the public key is specified as an ISO 8601 formatted string: YYYY-MM-DDTHH:MM:SSZ.
```
Comment (lbakerchef) [inline docs-chef-io/content/server/auth.md:400]:
```suggestion
    Name is an ASCII string used to identify the public key.  The expiration date for the public key is specified as an ISO 8601 formatted string: YYYY-MM-DDTHH:MM:SSZ.
```

File: docs-chef-io/content/server/auth.md [hunk 6]
```diff
@@ -386,6 +386,61 @@ common causes:
...
+    User can list, add, edit and delete public keys with knife user key command, below are the operations supported.
```
Comment (lbakerchef) [inline docs-chef-io/content/server/auth.md:401]:
```suggestion
    User can list, add, edit and delete public keys with knife user key command.  The following operations supported.
```

File: docs-chef-io/content/server/auth.md [hunk 7]
```diff
@@ -386,6 +386,61 @@ common causes:
...
+-  Using knife user key command
+    User can have multiple public keys associated with his chef server account.  Each public key has a name and expiration date associated with it.
+    Name is an ASCII string used to identify the public key.  The expiration date for the public key is specified as an ISO 8601 formatted string: YYYY-MM-DDTHH:MM:SSZ.
+    User can list, add, edit and delete public keys with knife user key command.  The following operations supported.
+        knife user key create to create public key.
+        knife user key delete to delete public key.
+        knife user key list to list user public keys.
+        knife user key edit to edit public keys.
+
+    Constraints: User can't edit key details which is currently in use.
```
Comment (IanMadd) [inline docs-chef-io/content/server/auth.md:407]: Does this just mean that if I'm using the `user1` key, I can't regenerate a new `user1` key? I have to create or use another key first and then regenerate user1 while using the other key?

Comment (IanMadd) [reply docs-chef-io/content/server/auth.md:407]: Does this also apply to the `knife user reregister` command?

File: docs-chef-io/content/server/auth.md [hunk 8]
```diff
@@ -386,6 +386,61 @@ common causes:
...
+
+       2) Create new key pair with the help of knife user key create.
+          #knife user key create user1 --key-name new_key_03_11_2022
+
+              {
+                "user": "user1",
+                "name": "new_key_03_11_2022",
+                "expiration_date": "infinity",
+                "create_key": true
+              }
```
Comment (IanMadd) [inline docs-chef-io/content/server/auth.md:422]: Is this generated data from knife? Does the user need to do something with it?

File: docs-chef-io/content/server/auth.md [hunk 9]
```diff
@@ -386,6 +386,61 @@ common causes:
...
+
+       3) Make new key effective by placing it in user private key file.
```
[the remark this answers is not in the document]
Comment (sreepuramsudheer) [reply docs-chef-io/content/server/auth.md:424]: adding a pem file to the .chef directory.

File: docs-chef-io/content/server/auth.md (hunk 8 above)
Comment (sreepuramsudheer) [reply docs-chef-io/content/server/auth.md:422]: This is knife generated content {generated with arguments passed to the knife command} which appears in a text editor where user can do modifications if required.

File: docs-chef-io/content/server/auth.md (hunk 7 above)
Comment (sreepuramsudheer) [reply docs-chef-io/content/server/auth.md:407]: I mean user can't edit details of the key which he is using to send current request.

**Does this also apply to the knife user reregister command?**
    No reregister command will always edit public key named default, create if one dose not exists.
    if current request is made with help of default private key then also request will succeed.

File: docs-chef-io/content/server/auth.md (hunk 8 above)
Comment (IanMadd) [reply docs-chef-io/content/server/auth.md:422]: Ok, does knife generate a file and place it somewhere when you run that command? Does this text already exist and the user has to update it, or does knife overwrite an existing file? I assume knife automatically opens the text editor, correct?

File: docs-chef-io/content/server/auth.md [hunk 10]
```diff
@@ -386,6 +386,77 @@ common causes:
...
+   knife user key list USERNAME
+   ```
+
+1. Create a new key pair:
+
+   ```sh
+   knife user key create USERNAME --key-name KEYNAME --expiration-date YYYY-MM-DDTHH:MM:SSZ
+   ```
+
+   Knife will open the `config.rb` or `credentials` file in your text editor. Modify the user credentials to match the newly created key pair name and expiration date.
```
Comment (sreepuramsudheer) [inline docs-chef-io/content/server/auth.md:430]: knife will not open config.rb or credentials file. it opens editor with request data which will be send in this operation. user can edit the request data before sending.

User can replace old key file with new key file, update config.rb or credentials file for the new private key to take effect.

Note: User can have multiple active key pairs and can choose one of the key {by configuring in config.rb or credentials}  to communicate with server.

Comment (IanMadd) [inline docs-chef-io/content/server/auth.md:430]: How about this?

```suggestion
   Knife will open your text editor with a data file containing the username, key name, and expiration date that will be sent to the Chef Infra Server. Modify the user credentials to match the newly created key pair name and expiration date.
```

File: docs-chef-io/content/server/auth.md [hunk 11]
```diff
@@ -386,6 +386,77 @@ common causes:
...
+
+   {{< note >}}
+
+   Specify the expiration date in ISO 8601 format.
+
+   The expiration date is optional. User keys don't expire if an expiration date isn't specified.
+
+   {{< /note >}}
+
```
Comment (IanMadd) [inline docs-chef-io/content/server/auth.md:439]:
```suggestion

1. Open your `config.rb` file or `credentials` file and modify it to match the new key name and expiration date.

```

**Review by vinay-satish — APPROVED** [review]

**Review by IanMadd — APPROVED** [review]

**Pull request merged**